src/Index.h

```diff
@@ -32,6 +32,18 @@ class Index {
     Index(Index &&other);
     ~Index();
 
+    struct IndexConfig {
+        bool numeric = false;
+        bool unique = false;
+        bool sparse = false;
+        bool indexLineOffsets = false;
+        IndexConfig(bool n, bool u, bool s, bool i) :
+                numeric{n}, unique{u}, sparse{s}, indexLineOffsets{i} {};
+
+        IndexConfig() :
+                numeric{false}, unique{false}, sparse{false}, indexLineOffsets{false} {};
```

Comment:
Maybe get rid of the "= false" on 36-39, or make this empty to default them all, else we're specifying the false in two places redundantly.

Also I'd be tempted to kill lines 40 and 41: this class is handy as it names all the fields. Having a constructor that lets you pass (true,false,false,true) just opens the door to the same error as before :)